pkg/graph/resolvers/model_plan.go

```diff
@@ -279,3 +282,118 @@ func ModelPlanNameHistory(logger *zap.Logger, modelPlanID uuid.UUID, sortDir mod
 
 	return nameHistory, nil
 }
+
+// ModelPlanShare implements resolver logic to share a model plan with a list of emails
+func ModelPlanShare(
+	ctx context.Context,
+	logger *zap.Logger,
+	store *storage.Store,
+	principal authentication.Principal,
+	emailService oddmail.EmailService,
+	emailTemplateService email.TemplateService,
+	modelPlanID uuid.UUID,
+	viewFilter *models.ModelViewFilter,
+	receiverEmails []string,
+	optionalMessage *string,
+) (bool, error) {
+	modelPlan, err := store.ModelPlanGetByID(logger, modelPlanID)
+	if err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			logger.Warn("No model plan found for the given modelPlanID",
+				zap.String("modelPlanID", modelPlanID.String()))
+			return false, fmt.Errorf("no model plan found for the given modelPlanID: %w", err)
+		}
+
+		return false, fmt.Errorf("failed to fetch the model plan: %w", err)
+	}
+
+	planBasics, err := PlanBasicsGetByModelPlanIDLOADER(ctx, modelPlanID)
+	if err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			logger.Warn("No plan basics found for the given modelPlanID",
+				zap.String("modelPlanID", modelPlanID.String()))
+			return false, fmt.Errorf("no plan basics found for the given modelPlanID: %w", err)
+		}
+
+		return false, fmt.Errorf("failed to fetch the plan basics: %w", err)
+	}
+
+	// Get client address
+	clientAddress := emailService.GetConfig().GetClientAddress()
+
+	// Get sender
+	sender := principal.Account().Email
+
+	// Get email template
+	emailTemplate, err := emailTemplateService.GetEmailTemplate(email.ModelPlanShareTemplateName)
+	if err != nil {
+		return false, fmt.Errorf("failed to get email template: %w", err)
+	}
+
+	// Get email subject
+	emailSubject, err := emailTemplate.GetExecutedSubject(email.ModelPlanShareSubjectContent{
+		UserName: principal.Account().CommonName,
+	})
+	if err != nil {
+		return false, fmt.Errorf("failed to execute email subject: %w", err)
+	}
+
+	var modelPlanCategories []models.ModelCategory
+	if planBasics.ModelCategory != nil {
+		modelPlanCategories = append(modelPlanCategories, *planBasics.ModelCategory)
+	}
```

Comment:
With the recent merge of the Model Category updates I think we should add the additional categories here as well (which is likely why it was configured to be a slice of Categories here!)